src/pow_c.c

```diff
@@ -374,6 +374,8 @@ static void *PoWC_getPoWContext(ImplContext *impl_ctx, int8_t *trytes, int mwm)
             memcpy(ctx->input_trytes, trytes, TRANSACTION_TRYTES_LENGTH);
             ctx->mwm = mwm;
             ctx->indexOfContext = i;
+            if (threads > 0 && threads < ctx->num_threads)
+                ctx->num_threads = threads;
```

Comment:
Should we reset `ctx->num_threads` when this PoW Context is freed?

Reply:
The scenario I was thinking is using `IRI` + `dcurl`.
Since the `--pow-threads` is assigned when activating the `IRI`,
the number of threads would be fixed during the execution time.
It would not have any problem.

However, if we use `dcurl` alone, it may come up with issues.
For example:
```
getPoWContext with thread number = N
...
freePoWContext
...
getPoWContext with thread number = N+1
```

I would like to reset `ctx->num_threads` by the pre-stored maximum thread value.
The stored value would be the new member in `struct _pow_XXX_context` called `num_max_threads`.
Or just call the function `get_avail_nprocs()` to reset it.

Reply:
No reset when the PoW Context is freed.
Just make sure the assigned thread number is in the reasonable range.